contracts/Erc20Quest.sol

```diff
@@ -22,41 +25,58 @@ contract Erc20Quest is Quest {
         uint256 questFee_,
         address protocolFeeRecipient_
     )
-    Quest(
-        rewardTokenAddress_,
-        endTime_,
-        startTime_,
-        totalParticipants_,
-        rewardAmountInWeiOrTokenId_,
-        questId_,
-        receiptContractAddress_
-    ) {
+        Quest(
+            rewardTokenAddress_,
+            endTime_,
+            startTime_,
+            totalParticipants_,
+            rewardAmountInWeiOrTokenId_,
+            questId_,
+            receiptContractAddress_
+        )
+    {
         questFee = questFee_;
         protocolFeeRecipient = protocolFeeRecipient_;
         questFactoryContract = QuestFactory(msg.sender);
     }
 
+    /// @dev Function that gets the maximum amount of rewards that can be claimed by all users. It does not include the protocol fee
+    /// @return The maximum amount of rewards that can be claimed by all users
     function maxTotalRewards() public view returns (uint256) {
         return totalParticipants * rewardAmountInWeiOrTokenId;
     }
 
+    /// @dev Function that gets the maximum amount of rewards that can be claimed by the protocol or the quest deployer
+    /// @return The maximum amount of rewards that can be claimed by the protocol or the quest deployer
     function maxProtocolReward() public view returns (uint256) {
-        return maxTotalRewards() * questFee / 10_000;
+        return (maxTotalRewards() * questFee) / 10_000;
     }
 
+    /// @notice Starts the quest
+    /// @dev Requires that the balance of the rewards in the contract is greater than or equal to the maximum amount of rewards that can be claimed by all users and the protocol
```

Comment:
we should also probably note here that it marks it as ready to start. It still needs to be past the start date

Reply:
Technically, we can call `start` before the start date (there's no check for it). Is that something we need to implement? I think we should also allow quest admins to start the quest whenever